core/templates/pages/contributor-dashboard-page/services/contribution-and-review.service.ts

```diff
@@ -16,158 +16,181 @@
  * @fileoverview Service for fetching and resolving suggestions.
  */
 
-require('domain/feedback_thread/FeedbackThreadObjectFactory.ts');
-require('domain/suggestion/suggestion.model.ts');
-require('domain/suggestion/SuggestionThreadObjectFactory.ts');
-require('pages/exploration-editor-page/services/exploration-data.service.ts');
-require('services/alerts.service.ts');
-
-require(
-  'pages/exploration-editor-page/exploration-editor-page.constants.ajs.ts');
-
-angular.module('oppia').factory('ContributionAndReviewService', [
-  '$http', 'UrlInterpolationService', 'ACTION_ACCEPT_SUGGESTION',
-  function(
-      $http, UrlInterpolationService, ACTION_ACCEPT_SUGGESTION) {
-    var _SUBMITTED_SUGGESTION_LIST_HANDLER_URL = (
-      '/getsubmittedsuggestions/<target_type>/<suggestion_type>');
-    var _REVIEWABLE_SUGGESTIONS_HANDLER_URL = (
-      '/getreviewablesuggestions/<target_type>/<suggestion_type>');
-    var _SUGGESTION_TO_EXPLORATION_ACTION_HANDLER_URL = (
-      '/suggestionactionhandler/exploration/<exp_id>/<suggestion_id>');
-    var _SUGGESTION_TO_SKILL_ACTION_HANDLER_URL = (
-      '/suggestionactionhandler/skill/<skill_id>/<suggestion_id>');
-    var _UPDATE_TRANSLATION_HANDLER_URL = (
-      '/updatetranslationsuggestionhandler/<suggestion_id>'
-    );
-    var _UPDATE_QUESTION_HANDLER_URL = (
-      '/updatequestionsuggestionhandler/<suggestion_id>'
-    );
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
+import { AppConstants } from 'app.constants';
+import { ContributionAndReviewBackendApiService }
+  from './contribution-and-review-backend-api.service';
+import { SuggestionBackendDict } from 'domain/suggestion/suggestion.model';
+import { StateBackendDict } from 'domain/state/StateObjectFactory';
+import { ImagesData } from 'services/image-local-storage.service';
+
+export interface OpportunityDict {
+  'skill_id': string;
+  'skill_description': string;
+}
+
+export interface FetchSuggestionsResponse {
+  'target_id_to_opportunity_dict': {
+    [propName: string]: OpportunityDict;
+  };
+  suggestions: SuggestionBackendDict[];
+}
+
+interface FetchSuggestionsReturn {
+  [propName: string]: {
+    suggestion: SuggestionBackendDict;
+    details: OpportunityDict;
+  };
+}
 
-    var _fetchSuggestionsAsync = async function(url) {
-      return $http.get(url).then(function(res) {
-        var suggestionIdToSuggestions = {};
-        var targetIdToDetails = res.data.target_id_to_opportunity_dict;
-        res.data.suggestions.forEach(function(suggestion) {
+@Injectable({
+  providedIn: 'root',
+})
+export class ContributionAndReviewService {
+  constructor(
+    private contributionAndReviewBackendApiService:
+      ContributionAndReviewBackendApiService
+  ) {}
+
+  private async _fetchSuggestionsAsync(
+      url: string, targetType: string, suggestionType: string
```

Comment:
Why the old and new function with the same name have different number of params?

Reply:
@DubeySandeep Previously, all the URL templates were present inside the `contribution-and-review-service` but now, they are moved into `contribution-and-review-backend-api-service`. If I move them back to the original service file, then we can have the structure similar to the old function. Then we have to pass the interpolated url into the `backend-api -service`. Should I do this? Also, the usage of this[url] will not be needed after that. Please let me know.

Reply:
I liked the approach of moving URLs to backend-api-services and it's consistent with our expected backend-api.ts file, how about having these two functions in the backend API service:
- fetchSubmittedSuggestionsAsync
- fetchReviewableSuggestionsAsync

Will it make it simpler and help us avoid `this[URL]` structure?

Reply:
Changed the param name of `_fetchSuggestionsAsync` to `fetchType` which can have values `SUBMITTED_QUESTION_SUGGESTIONS`, `REVIEWABLE_QUESTION_SUGGESTIONS`, `SUBMITTED_TRANSLATION_SUGGESTIONS` and `REVIEWABLE_TRANSLATION_SUGGESTIONS`. Also added the two suggested functions and removed the usage of `this[url]`.